### ISSUE TITLE
Support lower exponent formatting

### ISSUE THREAD

**Issue by ChristopherRabotin** [rustsim/nalgebra#653]
For scientific applications, printing vectors with [LowerExp](https://doc.rust-lang.org/std/fmt/trait.LowerExp.html) would be very useful.

Here is an example of the error one would get today:
```
39 |     println!("gradient = {:e}", grad);
   |                                 ^^^^ the trait `std::fmt::LowerExp` is not implemented for `na::Matrix<f64, na::U6, na::U6, na::ArrayStorage<f64, na::U6, na::U6>>`
```

**Comment by sebcrozet** [comment]
Hi! This has been implemented by #635 (and merged to the `dev` branch) but is not yet released.

**Comment by ChristopherRabotin** [comment]
Awesome!
